clib/noDupArray.ml

```diff
@@ -0,0 +1,62 @@
+(************************************************************************)
+(*         *      The Rocq Prover / The Rocq Development Team           *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+
+type 'a t = 'a option array * bool ref
```

Comment:
Why isn't this `'a option array option ref` with invalidate being `:= None`?

Reply:
I guess this would be faster

Reply:
I made the change, I rewrote it using let operators as well